common/app/views/support/cleaner/VideoEmbedCleaner.scala

```diff
@@ -57,22 +57,21 @@ case class VideoEmbedCleaner(article: Article) extends HtmlCleaner {
 
         val video = findVideoApiElement(mediaId)
 
+        // remove all sources as we add them later in a specific order
+        // similar to video.scala.html template
+        // see `common.app.model.EncodingOrdering` for order
         element.getElementsByTag("source").remove()
 
-        val sourceHTML: String = getVideoAssets(mediaId).map { videoAsset =>
-          videoAsset.encoding.map { encoding =>
-            s"""<source src="${encoding.url}" type="${encoding.format}"></source>"""
-          }.getOrElse("")
-        }.mkString("")
-
-        element.append(sourceHTML)
-
         // add the poster url
         video.map(_.images).flatMap(Item640.bestFor).map(_.toString()).foreach { url =>
           element.attr("poster", url)
         }
 
-        findVideoApiElement(mediaId).foreach { videoElement =>
+        video.foreach { videoElement =>
+          videoElement.videos.encodings.map { encoding => {
```

Comment:
I would have simply added an unit test for the `MediaVideo` class (to avoid the html parsing gymnastic) but yours does the job I guess.